checkmate/checker/url/blocklist.py

```diff
@@ -0,0 +1,122 @@
+"""The abstract blocklist object."""
+
+import fnmatch
+import os
+import re
+from logging import getLogger
+from urllib.parse import urlparse
+
+from checkmate.checker.url.reason import Reason
+
+
+class Blocklist:
+    """A blocklist which URLs can be checked against.
+
+    For details of how to change the blocklist see:
+      * https://stackoverflow.com/c/hypothesis/questions/102/250
+
+    And is downloaded locally, via supervisor using `bin/fetch-blocklist`
+    """
+
+    LOG = getLogger(__name__)
+
+    # viahtml is ok with video, as far as we can tell
+    PERMITTED = (Reason.MEDIA_VIDEO,)
+
+    def __init__(self, filename):
+        self.LOG.debug("Monitoring blocklist file '%s'", filename)
+
+        self._filename = filename
+        self._last_modified = None
+        self.domains = {}
+        self.patterns = {}
+
+        self._refresh()
+
+    def check_url(self, url):
```

Comment:
This method was `is_blocked` and only returned a single value, rather than potentially yielding many